Sources/WorkerCapabilities/ConcreteCapabilities/WorkerArchitectureCapabilitiesProvider.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  File.swift
+//
+//
+//  Created by a.smolianin on 15.07.2022.
+//
+
+import Foundation
+import WorkerCapabilitiesModels
+import ProcessController
+import FileSystem
+import DateProvider
+import EmceeLogging
+
+public final class WorkerArchitectureCapabilitiesProvider: WorkerCapabilitiesProvider {
+    private let logger: ContextualLogger
+    private let commandExecutor: CommandExecutor
+    
+    public init(logger: ContextualLogger, commandExecutor: CommandExecutor) {
+        self.logger = logger
+        self.commandExecutor = commandExecutor
+    }
+    
+    convenience public init(logger: ContextualLogger) {
+        self.init(logger: logger, commandExecutor: DefaultComandExecutor())
+    }
+    
+    public func workerCapabilities() -> Set<WorkerCapability> {
+        return Set([WorkerCapability(name: WorkerCapabilityName("emcee.arch"), value: "\(getArch())")])
+    }
+    
+    private func getArch() -> Arch {
+        var result = "\(Arch.unknown)"
+        do {
+            result = try commandExecutor.executeCommandWithOutput(env: [String: String](), arguments: ["/usr/sbin/sysctl", "-n", "machdep.cpu.brand_string"])
+            logger.info("Command \"sysctl -n machdep.cpu.brand_string\" = \(result)")
+        } catch {
+            logger.error("Command \"sysctl -n machdep.cpu.brand_string\" could not be completed. Error: \(error.localizedDescription)")
+            return .unknown
+        }
+        switch result {
+        case let arch where arch.contains(Arch.m1.rawValue):
+            return .m1
+        case let arch where arch.contains(Arch.x86.rawValue):
+            return .x86
+        default:
+            return .unknown
+        }
+    }
+    
+    enum Arch: String {
+        case m1 = "Apple M1"
+        case x86 = "Intel"
+        case unknown //for m2 .etc
```

Comment:
Did you want to expose CPU arch (x86_64, arm64), or CPU type, or CPU manufacturer? It seems currently these all are messed up a bit.

I assume it is more beneficial to have arch rather than manufacturer or kind (i.e. Core i5 or i7 or M1 or M2 Pro vs x86_64 and arm64). Could you please update your PR to expose architecture only?